packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1304,6 +1305,58 @@ export class MergeTree {
 		return undefined;
 	}
 
+	/**
+	 * Finds the nearest reference with ReferenceType.Tile to `startPos` in the direction dictated by `posPrecedesTile`.
+	 * Uses depthFirstNodeWalk in addition to block-accelerated functionality. The search position will be included in
+	 * the nodes to walk, so searching on all positions, including the endpoints, can be considered inclusive.
+	 * Any out of bound search positions will return undefined, so in order to search the whole string, a forward
+	 * search can begin at 0, or a bacward search can begin at length-1.
+	 *
+	 * @param startPos - Position at which to start the search
+	 * @param clientId - clientId dictating the perspective to search from
+	 * @param tileLabel - Label of the tile to search for
+	 * @param backwards - Whether the desired tile comes before (true) or after (false) `startPos`
+	 */
+	public searchForTile(startPos: number, clientId: number, tileLabel: string, forwards = true) {
+		const searchInfo: IReferenceSearchInfo = {
+			mergeTree: this,
+			forwards,
+			tileLabel,
+		};
+
+		const { segment } = this.getContainingSegment(startPos, UniversalSequenceNumber, clientId);
+		const segWithParent: IMergeLeaf = segment as IMergeLeaf;
+		if (segWithParent?.parent === undefined) {
+			return undefined;
+		}
+
+		depthFirstNodeWalk(
+			segWithParent.parent,
+			segWithParent,
+			(seg) => {
+				if (seg.isLeaf()) {
+					if (Marker.is(seg) && refHasTileLabel(seg, tileLabel)) {
+						searchInfo.tile = seg;
+					}
+				} else {
+					const block = <IHierBlock>seg;
+					const marker = searchInfo.forwards
+						? <Marker>block.leftmostTiles[tileLabel]
```

Comment:
are these always markers? should we assert Marker.is

Reply:
also, the current is the same as `as` casting, so should be avoided

Reply:
They're ReferencePositions, so they fail the Marker.is asserts. If we continue to return ReferencePositions, I can probably avoid casting, but if we want to return Marker | undefined, I think we would need to cast somewhere.